Flaky raptor tests: `.FileSystemException: /tmp/.... failed to delete one or more files; see suppressed exceptions for details`

```
java.io.UncheckedIOException: java.nio.file.FileSystemException: /tmp/TrinoTest3453139378186970036: failed to delete one or more files; see suppressed exceptions for details
	at io.trino.testing.DistributedQueryRunner.close(DistributedQueryRunner.java:611)
	at io.trino.util.AutoCloseableCloser.close(AutoCloseableCloser.java:52)
	at io.trino.testing.AbstractTestQueryFramework.close(AbstractTestQueryFramework.java:109)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:515)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:217)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:144)
	at org.testng.internal.TestMethodWorker.invokeAfterClassMethods(TestMethodWorker.java:217)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:115)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.nio.file.FileSystemException: /tmp/TrinoTest3453139378186970036: failed to delete one or more files; see suppressed exceptions for details
	at com.google.common.io.MoreFiles.throwDeleteFailed(MoreFiles.java:810)
	at com.google.common.io.MoreFiles.deleteRecursively(MoreFiles.java:561)
	at io.trino.server.testing.TestingTrinoServer.lambda$close$2(TestingTrinoServer.java:374)
	at com.google.common.io.Closer.close(Closer.java:218)
	at io.trino.server.testing.TestingTrinoServer.close(TestingTrinoServer.java:383)
	at com.google.common.io.Closer.close(Closer.java:218)
	at io.trino.testing.DistributedQueryRunner.close(DistributedQueryRunner.java:608)
	... 15 more
	Suppressed: java.nio.file.DirectoryNotEmptyException
		at java.base/sun.nio.fs.UnixSecureDirectoryStream.implDelete(UnixSecureDirectoryStream.java:204)
		at java.base/sun.nio.fs.UnixSecureDirectoryStream.deleteDirectory(UnixSecureDirectoryStream.java:221)
		at java.base/sun.nio.fs.UnixSecureDirectoryStream.deleteDirectory(UnixSecureDirectoryStream.java:42)
		at com.google.common.io.MoreFiles.deleteRecursivelySecure(MoreFiles.java:637)
		at com.google.common.io.MoreFiles.deleteRecursively(MoreFiles.java:538)
		... 20 more
```

--- a/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/backup/BackupManager.java
+++ b/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/backup/BackupManager.java
@@ -36,12 +36,14 @@ import java.util.concurrent.ExecutorService;
 import java.util.concurrent.atomic.AtomicInteger;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.util.concurrent.MoreExecutors.shutdownAndAwaitTermination;
 import static io.airlift.concurrent.Threads.daemonThreadsNamed;
 import static io.trino.plugin.raptor.legacy.RaptorErrorCode.RAPTOR_BACKUP_CORRUPTION;
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.CompletableFuture.completedFuture;
 import static java.util.concurrent.CompletableFuture.runAsync;
 import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.concurrent.TimeUnit.SECONDS;
 
 public class BackupManager
 {
@@ -72,7 +74,7 @@ public class BackupManager
     @PreDestroy
     public void shutdown()
     {
-        executorService.shutdownNow();
+        shutdownAndAwaitTermination(executorService, 10, SECONDS);
     }
 
     public CompletableFuture<Void> submit(UUID uuid, File source)

--- a/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/metadata/DatabaseMetadataModule.java
+++ b/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/metadata/DatabaseMetadataModule.java
@@ -86,7 +86,7 @@ public class DatabaseMetadataModule
         @ForMetadata
         public static ConnectionFactory createConnectionFactory(H2DatabaseConfig config)
         {
-            String url = format("jdbc:h2:%s;DB_CLOSE_DELAY=-1", config.getFilename());
+            String url = format("jdbc:h2:retry:%s", config.getFilename());
             return () -> DriverManager.getConnection(url);
         }
 

--- a/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/storage/ShardRecoveryManager.java
+++ b/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/storage/ShardRecoveryManager.java
@@ -171,7 +171,9 @@ public class ShardRecoveryManager
             }
         }
         catch (Throwable t) {
-            log.error(t, "Error creating shard recovery tasks");
+            if (!executorService.isShutdown()) {
+                log.error(t, "Error creating shard recovery tasks");
+            }
         }
     }
 

--- a/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/storage/ShardRecoveryManager.java
+++ b/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/storage/ShardRecoveryManager.java
@@ -57,6 +57,7 @@ import java.util.function.Consumer;
 
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static com.google.common.util.concurrent.MoreExecutors.shutdownAndAwaitTermination;
 import static io.airlift.concurrent.MoreFutures.addExceptionCallback;
 import static io.airlift.concurrent.Threads.daemonThreadsNamed;
 import static io.airlift.units.DataSize.succinctBytes;
@@ -136,8 +137,8 @@ public class ShardRecoveryManager
     @PreDestroy
     public void shutdown()
     {
-        executorService.shutdownNow();
-        missingShardExecutor.shutdownNow();
+        shutdownAndAwaitTermination(missingShardExecutor, 10, SECONDS);
+        shutdownAndAwaitTermination(executorService, 10, SECONDS);
     }
 
     private void scheduleRecoverMissingShards()

--- a/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/storage/organization/ShardOrganizer.java
+++ b/plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/storage/organization/ShardOrganizer.java
@@ -28,14 +28,15 @@ import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.ThreadPoolExecutor;
-import java.util.concurrent.atomic.AtomicBoolean;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.collect.Sets.newConcurrentHashSet;
+import static com.google.common.util.concurrent.MoreExecutors.shutdownAndAwaitTermination;
 import static io.airlift.concurrent.Threads.daemonThreadsNamed;
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.CompletableFuture.runAsync;
 import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.concurrent.TimeUnit.SECONDS;
 
 public class ShardOrganizer
 {
@@ -43,8 +44,6 @@ public class ShardOrganizer
 
     private final ExecutorService executorService;
     private final ThreadPoolExecutorMBean executorMBean;
-
-    private final AtomicBoolean shutdown = new AtomicBoolean();
 
     // Tracks shards that are scheduled for compaction so that we do not schedule them more than once
     private final Set<UUID> shardsInProgress = newConcurrentHashSet();
@@ -69,9 +68,7 @@ public class ShardOrganizer
     @PreDestroy
     public void shutdown()
     {
-        if (!shutdown.getAndSet(true)) {
-            executorService.shutdownNow();
-        }
+        shutdownAndAwaitTermination(executorService, 10, SECONDS);
     }
 
     public CompletableFuture<?> enqueue(OrganizationSet organizationSet)

